docs/pages/repo/docs/core-concepts/monorepos/task-dependencies.mdx

```diff
@@ -0,0 +1,189 @@
+---
+title: Task Dependencies
+description: Turborepo helps you specify task dependencies declaratively.
+---
+
+import Callout from "../../../../../components/Callout";
+import HeartIcon from "@heroicons/react/solid/HeartIcon";
+import { Tabs, Tab } from '../../../../../components/Tabs'
+
+# Task Dependencies
+
+Turborepo is most powerful when you express how your tasks relate to each other.
+We refer to these relationships as "dependencies", but they are not the same as
+package dependencies that you install from your `package.json` files. While Turborepo
+does understand your workspaces, it does not automatically draw any
+relationships between their tasks, unless you express them in `turbo.json` via
+the `dependsOn` configuration.
+
+Let's walk through some common patterns on how to make a task depend on other tasks.
+
+## From the same workspace
+
+There might be tasks that need to run _before_ other tasks. For instance,
+`build` might need to be run before `deploy`.
+
+If both tasks are in the same workspace, you can specify the relationship like
+this:
+
+```jsonc filename="turbo.json"
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    "build": {},
+    "deploy": {
+      // A workspace's `deploy` task depends on the `build` task of the same workspace.
+      "dependsOn": ["build"]
+    }
+  }
+}
+```
+
+This means that whenever `turbo deploy` is run, `build` will also be run
+inside the same workspace.
+
+## From dependent workspaces
+
+A common pattern in monorepos is to declare that a workspace's `build` task
+should only run once the `build` tasks of all _the workspaces it depends on_ are
+complete.
+
+<Callout type="info">
+This can be confusing as it refers to both _workspace_ dependencies
+and _task_ dependencies, which are different concepts. Workspace dependencies are `dependencies`
+and `devDependencies` in `package.json`, whereas, task dependencies are `dependsOn` key in `turbo.json`.
+</Callout>
+
+The `^` symbol (called a "caret") explicitly declares that the task depends on the task in a
+workspace it depends on.
+
+```jsonc filename="turbo.json"
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipeline": {
+    "build": {
+      // A workspace's `build` command depends on its dependencies'
+      // and devDependencies' `build` commands being completed first
+      "dependsOn": ["^build"],
+    }
+  }
+}
+```
+
+With the configuration above, if an app installs a package from another workspace, the package's
+`build` script will always run before the app's `build` script.
+
+## From arbitrary workspaces
+
+Sometimes, you may want a workspace-task depend on another
+workspace-task. This can be especially helpful for repos migrating from `lerna`
```

Comment:
```suggestion
Sometimes, you may want a workspace-task to depend on another
workspace-task. This can be especially helpful for repos migrating from `lerna`
```